randovania/game_connection/connector/am2r_remote_connector.py

```diff
@@ -0,0 +1,241 @@
+            None,
+        )
+        self.PlayerLocationChanged.emit(PlayerLocationEvent(self.current_region, None))
+
+    def new_collected_locations_received(self, new_indices: str) -> None:
+        """
+        __Collected locations implementation detail:__
+        The game keeps track of which locations it collected, in a big string, with the format `locations:XX,YY,ZZ...`,
+        where XX/YY/ZZ are the pickup indices. The pickup indices match between game and DB.
+        Also, this string is saved to the save file.
+        The game sends that string to RDV, where we then loop through it, and mark every index as collected.
+
+        :param new_indices: A string from the game with the format `locations:XX,YY,ZZ...`.
+        """
+
+        locations = set()
+        start_of_indices = "locations:"
+
+        if not new_indices.startswith(start_of_indices):
+            self.logger.warning("Unknown response: %s", new_indices)
+            return
+
+        for index in new_indices[len(start_of_indices) :].split(","):
+            if not index.isdigit():
+                self.logger.warning("Response should contain a digit, but instead contains '%s'", index)
+                continue
+            locations.add(PickupIndex(int(index)))
+
+        for location in locations:
+            self.PickupIndexCollected.emit(location)
+
+    def new_inventory_received(self, new_inventory: str) -> None:
+        """
+        __New inventory implementation detail:__
+        The game keeps track of which items it collected in a big string with the format
+        `items:XA|XB,YA|YB,ZA|ZB...`, where XA/YA/ZA are the item names and XB/YB/ZB is the quantity of that item.
+        This string is also saved to the save file. The item names *mostly* match up with the RDV item names,
+        except for ammo and progressives.
+        The game sends that string to RDV, where we then loop through it. We adjust the item name for ammo and
+        progressives accordingly, and stuff it into an item|quantity dict.
+        If the item exists in that dict, we increase its quantity, if not, we add it. Then, based on that dict, we
+        update the inventory.
+
+        Regarding ammo and progressives: for some reason, we need here the name of the ammo, and not the name of the
+        expansions. I.e. for `Missile Expansions`, we need `Missiles`. So there's a lookup dict here to adjust for them.
+
+        :param new_inventory: A string from the game with the format `items:XA|XB,YA|YB,ZA|ZB...`.
+        """
+        locations: dict[ItemResourceInfo, int] = {}
+        start_of_inventory = "items:"
+
+        if not new_inventory.startswith(start_of_inventory):
+            self.logger.warning("Unknown response: %s", new_inventory)
+            return
+
+        for position in new_inventory[len(start_of_inventory) :].split(","):
+            print(bytes(position, "utf-8"))
```

Comment:
Should be removed later or properly logged.

Reply:
forgot to remove, whoops